src/utils/http/fetchAPI.js

```diff
@@ -0,0 +1,56 @@
+import APIError from 'utils/http/errors/APIError';
+import getApiEndpoint from '../apiEndpoint';
+import getAuthJWT from '../getAuthJWT';
+import FetchError from './errors/FetchError';
+
+const fetchAPI = async (path, params = {}, extras = {}, returnValue = true) => {
+  const headers = params.headers ? params.headers : {};
+  const authJWT = extras.jwt || await getAuthJWT();
+
+  const parameters = {
+    ...params,
+    headers: {
+      ...headers,
+      ...(authJWT && { Authorization: `Bearer ${authJWT}` }),
+    },
+  };
+
+  const url = getApiEndpoint(extras.uiUrl) + path;
+
+  let response;
+  try {
+    response = await fetch(url, parameters);
+  } catch (e) {
+    // wrap fetch errors in custom error
+    throw new FetchError(e);
+  }
+
+  const isJson = response.headers.get('content-type')?.includes('application/json');
+
+  if (response.ok) {
```

Comment:
Do we still need this clause? At the end of the function you also return `response.json()`

Reply:
it's used only in the case of an error. Ideally, it should be used for both cases, however most of the tests do not set that field so using the `fetchAPI` with those will fail. In the general situation, when we want `fetchAPI` to return something it's safe to assume it's JSON. But when there's an error there might not be json which would lead to `response.json()` to fail, and throw an exception hiding the real exception. That's why in that case I only deserialize json when I'm sure it won't fail.